src/main/java/com/iexec/common/utils/TargetedLock.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.utils;
+
+import net.jodah.expiringmap.ExpiringMap;
+
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Provides a way to avoid parallel runs of an action.
+ * Locks are based on a key, so that two actions with the same key can't be run at the same time
+ * but two actions with different keys can be run at the same time.
+ * <br>
+ * To avoid memory-leaks, entries expire after a given time.
+ * Default is 1 hour, but it can be set at {@link TargetedLock} object creation.
+ *
+ * @param <K> Type of the key.
+ */
+public class TargetedLock<K> {
```

Comment:
```suggestion
public class LockRunner<K> {
```
?

Reply:
I'd prefer to keep a reference to `target` as we don't lock everything, just what's related to a target/key.

Reply:
`TargetedLockRunner<K>`
?
(A lock is usually binary/passive, but this class does more since it performs some actions)

Reply:
I agree with you. `TargetedLockRunner` seems better, let's continue with that! Thanks :)